include/eve/arch/riscv/rvv_common_masks.hpp

```diff
@@ -0,0 +1,56 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/function/friends.hpp>
+
+namespace eve::detail
+{
+template<arithmetic_scalar_value T, typename N>
+EVE_FORCEINLINE logical<wide<T, N>>
+                rvv_true()
+{
+  static constexpr auto   lmul     = riscv_rvv_dyn_::getLMUL<T>(N::value);
+  static constexpr size_t size     = sizeof(T) * 8;
+  static constexpr size_t bit_size = lmul > 0 ? size / lmul : size * (-lmul);
+  if constexpr( bit_size == 1 ) return __riscv_vmset_m_b1(N::value);
+  if constexpr( bit_size == 2 ) return __riscv_vmset_m_b2(N::value);
+  if constexpr( bit_size == 4 ) return __riscv_vmset_m_b4(N::value);
+  if constexpr( bit_size == 8 ) return __riscv_vmset_m_b8(N::value);
+  if constexpr( bit_size == 16 ) return __riscv_vmset_m_b16(N::value);
+  if constexpr( bit_size == 32 ) return __riscv_vmset_m_b32(N::value);
+  if constexpr( bit_size == 64 ) return __riscv_vmset_m_b64(N::value);
+}
+
+template<arithmetic_scalar_value T, typename N>
+EVE_FORCEINLINE logical<wide<T, N>>
+                rvv_none()
+{
+  static constexpr auto   lmul     = riscv_rvv_dyn_::getLMUL<T>(N::value);
+  static constexpr size_t size     = sizeof(T) * 8;
+  static constexpr size_t bit_size = lmul > 0 ? size / lmul : size * (-lmul);
+  if constexpr( bit_size == 1 ) return __riscv_vmclr_m_b1(N::value);
```

Comment:
my lack of knowledge here shows through heavily - what is the difference between different clears? I'd expect all clears to be the same.

Reply:
They return different types.

For example, `__riscv_vmclr_m_b1` returns `vbool1_t`, and `__riscv_vmclr_m_b8` results in `vbool8_t`.